airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/unstructured_parser.py

```diff
@@ -91,25 +114,89 @@ def parse_records(
                 }
 
     def _read_file(self, file_handle: IOBase, remote_file: RemoteFile, format: UnstructuredFormat, logger: logging.Logger) -> Optional[str]:
-        _import_unstructured()
-        if (
-            (not unstructured_partition_pdf)
-            or (not unstructured_partition_docx)
-            or (not unstructured_partition_pptx)
-            or (not unstructured_optional_decode)
-        ):
-            # check whether unstructured library is actually available for better error message and to ensure proper typing (can't be None after this point)
-            raise Exception("unstructured library is not available")
-
         filetype = self._get_filetype(file_handle, remote_file)
 
         if filetype == FileType.MD:
             file_content: bytes = file_handle.read()
-            decoded_content: str = unstructured_optional_decode(file_content)
+            decoded_content: str = optional_decode(file_content)
             return decoded_content
         if filetype not in self._supported_file_types():
-            self._handle_unprocessable_file(remote_file, format, logger)
+            self._handle_unprocessable_file(remote_file, format.skip_unprocessable_file_types, logger)
             return None
+        if format.processing.mode == "local":
+            return self._read_file_locally(file_handle, filetype)
+        elif format.processing.mode == "api":
+            result: Optional[str] = self._read_file_remotely_with_retries(file_handle, format.processing, filetype)
```

Comment:
Would you do this for all kinds of exceptions of only 4xx user errors? Other problems are not necessarily config errors strictly speaking, even though they might be. For example if the unstructured API isn't available for a few minutes, retrying the whole sync wouldn't be the worst thing in the world to do.

Also OK with being pragmatic here and just treating everything as config error as this is an experimental feature.

Reply:
I'm in favor of raising config errors for all HTTP error codes since we can't do anything to solve either 4XX or 5XX but assuming all exceptions are config errors doesn't sound right since there could be other issues in the parser.

@flash1293 always raising config errors because this is experimental is also fine but that reason should be explicit either as a comment in the code or in the error message